handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -494,6 +494,12 @@ public void flush(ChannelHandlerContext ctx) throws Exception {
         }
         try {
             wrap(ctx, false);
+        } catch (Exception e) {
+            // Just fail the pending writes in case of an SSLException. If there are no pending writes rethrow.
+            if (pendingUnencryptedWrites.removeAndFailAll(e)) {
```

Comment:
IMO the situation in `write` is a bit different because there is no queue and the user is directly passing in a promise ... and so it is expected this promise will be used to reflect the status of the operation. My concern is that conditionally throwing here may lead to code paths that are not commonly taken but may occur infrequently in production and result in unexpected behavior ... this is why I would like to understand if there is a concrete use case we had in mind here so I can better understand.